Added Dashboard view for Infrastructure views.

h-kataria opened the pull request:
Added a button to toggle between Dashboard view and regular summary screen for Infrastructure providers.

screenshot:
![infra_provider_dashboard](https://cloud.githubusercontent.com/assets/3450808/19179465/9d532644-8c2b-11e6-80fe-a6e74f4d44d4.png)


Pivotal link: https://www.pivotaltracker.com/story/show/129629067

martinpovolny commented on app/controllers/ems_infra_dashboard_controller.rb:
```diff
@@ -0,0 +1,36 @@
+
+  def index
+    redirect_to :action => 'show'
+  end
```
@h-kataria : why do we need both `show` and `index`. I get that it's a legacy stuff on some controllers but why does it have to be in new ones?

martinpovolny commented on app/controllers/ems_infra_dashboard_controller.rb:
```diff
@@ -0,0 +1,36 @@
+
+  private
+
+  def get_session_data
```
same with `{get|set}_session_data` can we just avoid these methods in new controllers?

martinpovolny commented on config/routes.rb:
```diff
@@ -1025,6 +1025,13 @@
+    :ems_infra_dashboard      => {
+      :get => %w(
+        show
+        data
```
see, there's even no routable `index` action here

martinpovolny commented:
Overall, pretty cool contribution!

martinpovolny commented on app/views/ems_infra_dashboard/show.html.haml:
```diff
@@ -0,0 +1 @@
+= render :partial => "shared/views/show_containers_dashboard"
```
Why do we need partials that just render other partials?

And why do we need a non-partial view that just calls a partial. That is 2 extra files :-(

That's not a new problem here, it's in the older code too:
```
devil - [~/Projects/manageiq] (master)$ grep -r show_containers_dashboard app/
app/views/container_dashboard/show.html.haml:= render :partial => "shared/views/show_containers_dashboard"
app/views/ems_container/_show_dashboard.html.haml:= render :partial => "shared/views/show_containers_dashboard"
```

h-kataria replied:
@martinpovolny addressed above comment.

jrafanie commented:
Restarting the build

h-kataria commented:
@martinpovolny please re-review. Addressed your previous comments and added Recent Hosts/VMs charts to dashboard screen, updated screenshot in the original PR comment.

h-kataria commented:
@dclarizio please review.

chessbyte commented:
Euwe Backport details:
```sh
$ git log
commit 7193fb86f8dd791fd257c447a16dc299fe51aef2
Author: Martin Povolny <mpovolny@redhat.com>
Date:   Mon Oct 10 18:55:10 2016 +0200

    Merge pull request #11643 from h-kataria/provider_dashboard
    
    Added Dashboard view for Infrastructure views.
    (cherry picked from commit b6e3cd201ead0416499bbcae6c4df795d6f68beb)
```
